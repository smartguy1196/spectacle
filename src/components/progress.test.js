import Progress from "./progress";
import React from "react";
import { mount } from "enzyme";
import { mountToJson } from "enzyme-to-json";

const _items = [ <div />, <div />, <div /> ];

describe("<Progress />", () => {
  test("should render PacMan correctly", () => {
    const context = { styles: { progress: { pacman: [] } } };
    const wrapper = mount((
      <Progress
        type="pacman"
        items={_items}
        currentSlide={2}
      />
    ), { context });
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  test("should render the number style correctly", () => {
    const context = { styles: { progress: { number: [] } } };
    const wrapper = mount((
      <Progress
        type="number"
        items={_items}
        currentSlide={1}
      />
    ), { context });
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  test("should render the bar style correctly", () => {
    const context = { styles: { progress: { bar: [] } } };
    const wrapper = mount((
      <Progress
        type="bar"
        items={_items}
        currentSlide={1}
      />
    ), { context });
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  test("should render nothing when none is provided.", () => {
    const context = { styles: { progress: {} } };
    const wrapper = mount((
      <Progress
        type="none"
        items={_items}
        currentSlide={3}
      />
    ), { context });
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
});
