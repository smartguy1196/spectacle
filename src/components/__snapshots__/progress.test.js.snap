exports[`<Progress /> should render PacMan correctly 1`] = `
<Progress
  currentSlide={2}
  items={
    Array [
      <div />,
      <div />,
      <div />,
    ]
  }
  type="pacman">
  <div
    data-radium={true}
    style={Object {}}>
    <div
      data-radium={true}>
      <div
        data-radium={true}
        style={
          Object {
            "left": "15px",
            "top": "-20px",
          }
        }>
        <div
          data-radium={true}
          style={
            Object {
              "animation": "pacmanTopFramesBis 0.12s linear 10 alternate both",
            }
          } />
        <div
          data-radium={true}
          style={
            Object {
              "animation": "pacmanBottomFramesBis 0.12s linear 10 alternate both",
            }
          } />
      </div>
      <div
        data-radium={true}
        style={
          Object {
            "left": "-25px",
            "opacity": 0,
            "top": "-20px",
          }
        } />
      <div
        data-radium={true}
        style={
          Object {
            "left": "-5px",
            "opacity": 0,
            "top": "-20px",
          }
        } />
      <div
        data-radium={true}
        style={
          Object {
            "left": "15px",
            "opacity": 0,
            "top": "-20px",
          }
        } />
    </div>
  </div>
</Progress>
`;

exports[`<Progress /> should render nothing when none is provided. 1`] = `
<Progress
  currentSlide={3}
  items={
    Array [
      <div />,
      <div />,
      <div />,
    ]
  }
  type="none" />
`;

exports[`<Progress /> should render the bar style correctly 1`] = `
<Progress
  currentSlide={1}
  items={
    Array [
      <div />,
      <div />,
      <div />,
    ]
  }
  type="bar">
  <div
    data-radium={true}
    style={Object {}}>
    <div
      data-radium={true}
      style={
        Object {
          "width": "50%",
        }
      } />
  </div>
</Progress>
`;

exports[`<Progress /> should render the number style correctly 1`] = `
<Progress
  currentSlide={1}
  items={
    Array [
      <div />,
      <div />,
      <div />,
    ]
  }
  type="number">
  <div
    data-radium={true}
    style={Object {}}>
    <div
      data-radium={true} />
  </div>
</Progress>
`;
